### PR TITLE
[O2B-1123] Use generic overview model in runs model

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Refactor RunsOverviewModel to use OverviewPageModel

Changes made to the database:
- NA

### CODE REVIEW

File: lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js [hunk 1]
```diff
@@ -18,25 +18,29 @@ import { runsActiveColumns } from '../ActiveColumns/runsActiveColumns.js';
...
+    const loadingData = !runs;
+    const exportEnabled = selectedRunsFields.length > 0 && !loadingData;
+
+    // eslint-disable-next-line require-jsdoc
+    const optionalTruncatedInfo = () => runsOverviewModel.areExportableItemsTruncated
+        ? h(
+            '#truncated-export-warning.warning',
+            `The runs export is limited to ${runs.length} entries, only the last runs will be exported (sorted by run number)`,
+        )
+        : null;
```
Comment (xsalonx) [inline lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js:40]: This message is not render immediately, I think during `loadingData`, spinner might be here instead

File: lib/public/views/Runs/Overview/RunsOverviewModel.js [hunk 2]
```diff
@@ -231,9 +211,9 @@ export class RunsOverviewModel extends Observable {
             || this.environmentIdsFilter !== ''
             || this.runQualitiesFilters.length !== 0
             || this._triggerValuesFilters.size !== 0
-            || this.nDetectorsFilter !== null
-            || this._nEpnsFilter !== null
-            || this.nFlpsFilter !== null
+            || this.nDetectorsFilter?.operator && this.nDetectorsFilter.limit
+            || this._nEpnsFilter?.operator && this._nEpnsFilter?.limit
+            || this.nFlpsFilter?.operator && this.nFlpsFilter?.limit
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/RunsOverviewModel.js:216]: Do not belong to this PR

File: lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js (hunk 1 above)
Comment (martinboulais) [inline lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js:40]: Please, stop disable eslint. In particular when the function name is not really explicit.
I don't see what extracting this function brings here, especially if it's at the same level as where the code were called before.

File: lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js [hunk 3]
```diff
@@ -97,15 +101,15 @@ const errorDisplay = () => h('.danger', 'Data fetching failed');
...
+    const runsRemoteData = runsOverviewModel.exportableItems;
 
     return h('div#export-runs-modal', [
         h('h2', 'Export Runs'),
         runsRemoteData.match({
-            NotAsked: () => errorDisplay(),
             Loading: () => exportForm(runsOverviewModel, null, modalHandler),
             Success: (payload) => exportForm(runsOverviewModel, payload, modalHandler),
-            Failure: () => errorDisplay(),
+            Other: () => errorDisplay(),
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js:112]: Are you sure this is a good idea? I think `NotAsked` displaying an error was a mistake, and you are hiding it here. `Other` should not lead to some error display.

File: lib/public/views/Runs/Overview/RunsOverviewModel.js [hunk 4]
```diff
@@ -51,81 +52,59 @@ export class RunsOverviewModel extends Observable {
...
 
-        this._pagination = new PaginationModel();
-        this._pagination.observe(() => this.fetchRuns());
-        this._pagination.itemsPerPageSelector$.observe(() => this.notify());
-
-        // Content
-        this._currentPageRuns = RemoteData.notAsked();
-        this._allRuns = RemoteData.notAsked();
+        // Exportable items
+        this._exportableDataSource = new PaginatedRemoteDataSource();
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/RunsOverviewModel.js:56]: Why is it paginated?

File: lib/public/views/Runs/Overview/RunsOverviewModel.js [hunk 5]
```diff
@@ -798,38 +778,42 @@ export class RunsOverviewModel extends Observable {
...
+     * Update the cache containing all the runs without paging
      *
-     * @return {RemoteData} the remote data of the runs
+     * @return {Promise<void>} void
+     * @private
      */
-    get allRuns() {
-        if (this._allRuns.isNotAsked()) {
-            this._fetchAllRunsWithoutPagination();
+    async loadExportableRuns() {
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/RunsOverviewModel.js:795]: You are completely breaking the previous logic here, are you aware of it?
Before, we fetch all runs only if the current list of runs can't be reused as export.

File: lib/public/views/Runs/Overview/RunsOverviewModel.js [hunk 6]
```diff
@@ -798,38 +778,42 @@ export class RunsOverviewModel extends Observable {
...
+        const endpoint = this.getRootEndpoint();
+        if (this._exportableItemsEndpoint === endpoint &&
+            (this.exportableItems.isSuccess() || this.exportableItems.isLoading())) {
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/RunsOverviewModel.js:798]: You are using deprecated code in new code, you should not do that.

File: lib/public/views/Runs/Overview/RunsOverviewModel.js [hunk 7]
```diff
@@ -51,81 +52,59 @@ export class RunsOverviewModel extends Observable {
...
-        this._pagination = new PaginationModel();
-        this._pagination.observe(() => this.fetchRuns());
-        this._pagination.itemsPerPageSelector$.observe(() => this.notify());
-
-        // Content
-        this._currentPageRuns = RemoteData.notAsked();
-        this._allRuns = RemoteData.notAsked();
+        // Exportable items
+        this._exportableDataSource = new PaginatedRemoteDataSource();
+        this._exportableItems = ObservableData.builder()
```
Comment (martinboulais) [inline lib/public/views/Runs/Overview/RunsOverviewModel.js:57]: For me it should simply be `exportItems`, because it's the items of the export.

**Review by martinboulais — CHANGES_REQUESTED** [review]

**Comment by martinboulais** [comment]
I think you are doing too much things here: 
- Update filtering 
- Use OverviewModel
- Use data source for runs export

This PR should only contain code change for the 2nd point, please update accordingly

File: lib/public/views/Runs/Overview/RunsOverviewModel.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/RunsOverviewModel.js]: revoked

File: lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js (hunk 1 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js]: changes, revoked

File: lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js (hunk 3 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/exportRunsTriggerAndModal.js:109]: revoked, if using `NotAsked` was an error, changing it doesn't belong to this PR...

File: lib/public/views/Runs/Overview/RunsOverviewModel.js (hunk 4 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/RunsOverviewModel.js]: bad idea, change to just `RemoteDataSource` with proper `RemoteData#apply`

File: lib/public/views/Runs/Overview/RunsOverviewModel.js (hunk 6 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/RunsOverviewModel.js]: refactored to use match

File: lib/public/views/Runs/Overview/RunsOverviewModel.js (hunk 7 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/RunsOverviewModel.js]: appied

File: lib/public/views/Runs/Overview/RunsOverviewModel.js (hunk 5 above)
Comment (xsalonx) [reply lib/public/views/Runs/Overview/RunsOverviewModel.js]: and it is so now, you can't reuse iff you changed filtering and this iff `root endpoint` is different - see condition if `loadExportRuns` (previously: `loadExportableRuns`)

**Review by martinboulais — APPROVED** [review]
